cpp/arcticdb/processing/test/test_has_valid_type_promotion.cpp

```diff
@@ -121,3 +121,28 @@ TEST(HasValidTypePromotion, FloatFloat) {
     EXPECT_FALSE(has_valid_type_promotion(float64, float32));
 }
 
+TEST(HasValidTypePromotion, EmptyToEverything) {
+    using namespace arcticdb;
+    using namespace arcticdb::entity;
+    TypeDescriptor source(ValueType::EMPTY, SizeBits::S64, Dimension::Dim0);
+    for(int value_type = int(ValueType::UNKNOWN_VALUE_TYPE); value_type < int(ValueType::COUNT); ++value_type) {
+        for(int size_bits = int(SizeBits::UNKNOWN_SIZE_BITS); size_bits < int(SizeBits::COUNT); ++size_bits) {
+            const TypeDescriptor target(ValueType(value_type), SizeBits(size_bits), Dimension::Dim0);
+            ASSERT_EQ(has_valid_type_promotion(source, target), target);
+            ASSERT_TRUE(trivially_compatible_types(source, target));
+        }
+    }
+}
+
+TEST(HasValidTypePromotion, EverythingToEmpty) {
+    using namespace arcticdb;
+    using namespace arcticdb::entity;
+    const TypeDescriptor target(ValueType::EMPTY, SizeBits::S64, Dimension::Dim0);
+    for(int value_type = int(ValueType::UNKNOWN_VALUE_TYPE); value_type < int(ValueType::COUNT); ++value_type) {
+        for(int size_bits = int(SizeBits::UNKNOWN_SIZE_BITS); size_bits < int(SizeBits::COUNT); ++size_bits) {
+            const TypeDescriptor source(ValueType(value_type), SizeBits(size_bits), Dimension::Dim0);
+            ASSERT_EQ(has_valid_type_promotion(source, target), target);
```

Comment:
This is wrong, `has_valid_type_promotion` should always return `std::nullopt` if the target is the empty type